ckanext/canada/tables/grants.yaml

```diff
@@ -666,84 +521,173 @@ resources:
-        •résiliations;
-        •remboursement.
-    obligation: Optional
+      en: The field indicates whether the intended beneficiaries belong to a specific demographic group.
+      fr: Le champ indique si les bénéficiaires visés font partie d’un groupe démographique identifié.
+    obligation: Mandatory
+    occurrence: Repeatable
+    excel_required: true
+    form_required: true
+    excel_full_text_choices: true
+    format_type: Controlled List
+    validation:
+      en: This field must not be empty.
+      fr: Ce champ ne doit pas être vide.
+    datastore_type: _text
+    form_snippet: scheming/form_snippets/text.html
+    choice_order_expression: 'value.get("order", False), code'
+    choices:
+      Y:
+        en: Youth
+        fr: Jeunes
+        order: 1
+      S:
+        en: Senior
+        fr: Personnes âgées
+        order: 2
+      D:
+        en: Persons with disability
+        fr: Personnes handicappées
+        order: 3
+      I:
+        en: Indigenous people
+        fr: Personnes autochtones
+        order: 4
+      V:
+        en: Visible minority
+        fr: Minorités visibles
+        order: 5
+      W:
+        en: Women
+        fr: Femmes
+        order: 6
+      L:
+        en: 2SLGBTQI+
+        fr: 2ELGBTQI+
+        order: 7
+      O:
+        en: Other
+        fr: Autre
+        order: 8
+      N:
+        en: None
+        fr: Aucun
+        order: 9
+
+  # 3.27 Other Demographic of intended beneficiary Description (English)
+  - datastore_id: other_dem_intended_beneficiary_description_en
+    label:
+      en: Demographic of intended beneficiary Description (English)
+      fr: Description du groupe démographique des bénéficiaires visés (anglais)
+    description:
+      en: >
+        This field is mandatory if “Other” is chosen as an
+        option for the field “Intended Beneficiary demographic
+        group”. This field has a maximum length of 1800
+        characters. This text must be provided in both languages.
+      fr: >
+        TBD
+        Ce champ ne peut excéder une longueur maximale de 1800
+        caractères. Ce texte doit être fourni en anglais et en français.
+    obligation:
+      en: This field is mandatory if “Other” is chosen as an option for the field “Intended Beneficiary demographic group”.
+      fr: TBD
+    excel_required: false
+    form_required: false
+    excel_column_width: 80
+    validation:
+      en: This field has a maximum length of 1800 characters. This text must be provided in both languages.
+      fr: Ce champ ne peut excéder une longueur maximale de 1800 caractères. Ce texte doit être fourni en anglais et en français.
+    occurrence: Single
     format_type: Free text
     datastore_type: text
-    form_snippet: scheming/form_snippets/textarea.html
+    max_chars: 1800
+    excel_required_formula: 'OR(TRIM(LEFT({dem_intended_beneficiary},FIND(":",{dem_intended_beneficiary}&":")-1))="O",NOT(ISBLANK({other_dem_intended_beneficiary_description_fr})))'
```

Comment:
@RabiaSajjad were you able to test this in excel with lower case? and if that is then accepted into the upload form as well?